Student/Student/Submissions/ArcSubmission/ArcSubmissionViewController.swift

```diff
@@ -49,8 +49,27 @@ class ArcSubmissionViewController: UIViewController, ArcSubmissionView {
 
         addCancelButton(side: .left)
 
-        webView.navigationDelegate = self
         spinner.startAnimating()
+
+        let config = WKWebViewConfiguration()
+        config.processPool = CoreWebView.processPool
+        webView = WKWebView(frame: .zero, configuration: config)
+        webView.navigationDelegate = self
+        webView.addScript(js)
+        webView.handle("submit") { [weak self] message in
+            self?.presenter?.submit(form: message.body) { error in
```

Comment:
`message.body as? [String: Any]` should work. You shouldn't need to use `JSONSerilaization` to get a dict representation of the message.

Reply:
I refactored this.